src/status_im/contexts/onboarding/events.cljs

```diff
@@ -1,31 +1,192 @@
-         [:dispatch [:profile/toggle-testnet-mode-banner]]
-         [:dispatch [:universal-links/process-stored-event]]]}))
+ :onboarding/biometrics-fail
+ (fn [_ [error]]
+   {:dispatch [:biometric/show-message (ex-cause error)]}))
 
-(rf/defn enable-biometrics
-  {:events [:onboarding/enable-biometrics]}
-  [_]
+(defn enable-biometrics
+  [_ [{:keys [on-done]}]]
   {:fx [[:dispatch
          [:biometric/authenticate
-          {:on-success #(rf/dispatch [:onboarding/biometrics-done])
-           :on-fail    #(rf/dispatch [:onboarding/biometrics-fail %])}]]]})
+          {:on-success #(rf/dispatch on-done)
+           :on-fail    #(rf/dispatch (conj on-done %))}]]]})
+
+(rf/reg-event-fx :onboarding/enable-biometrics enable-biometrics)
+
+;;; -------
+
+(defn inject-local-profile-storage
+  [context]
+  (let [db            (interceptor/get-coeffect context :db)
+        key-uid       (get-in db [:profile/profile :key-uid])
+        local-profile (mmkv/get-object key-uid)]
+    (assoc-in context
+     [:coeffects :local-profile-storage]
+     local-profile)))
+
+(def local-profile-storage-interceptor
+  (interceptor/->interceptor
+   :id     :local-profile-storage-interceptor
+   :before inject-local-profile-storage))
+
+(rf/reg-fx :effects.profile/save-notifications-prompted
+ (fn [{:keys [key-uid]}]
+   (let [profile-storage (mmkv/get-object key-uid {})]
+     (mmkv/set-object key-uid
+                      (assoc profile-storage :notifications-prompted? true)))))
+
+(rf/reg-fx :effects.profile/remove-local-profile-storage
+ (fn [{:keys [key-uid]}]
+   (mmkv/delete-key key-uid)))
+
+(rf/reg-event-fx :shell/show-root-view
+ [local-profile-storage-interceptor]
+ (fn [{:keys [db local-profile-storage]} [{:keys [notifications-prompt-skip?]}]]
+   (let [{:keys [key-uid]}  (get-in db [:profile/profile])
+         onboarding-profile (get-in db [:onboarding/profile])]
+     (if (ff/enabled? ::ff/settings.news-notifications)
+       (if (or (:notifications-prompted? local-profile-storage)
+               (:notifications-prompted? onboarding-profile)
+               notifications-prompt-skip?)
+         {:fx (cond-> []
+                (not (:notifications-prompted? local-profile-storage))
+                (conj [:effects.profile/save-notifications-prompted
+                       {:key-uid key-uid}])
+                :else
+                (conj [:dispatch [:update-theme-and-init-root :screen/shell-stack]]
+                      [:dispatch [:profile/toggle-testnet-mode-banner]]))}
+         {:fx [[:effects.profile/save-notifications-prompted {:key-uid key-uid}]
+               [:dispatch
+                [:onboarding/notifications-setup
+                 {:key-uid     key-uid
+                  :biometrics? false
+                  :syncing?    false
+                  :onboarding? false}]]]})
+       {:fx [[:dispatch [:update-theme-and-init-root :screen/shell-stack]]
+             [:dispatch [:profile/toggle-testnet-mode-banner]]]}))))
```

Comment:
This event (`:shell/show-root-view`) is a new event that checks if the enable-notifications screen has been shown for a profile. If not then the enable-notifications screen will be shown, but if it has already been showed then the app will display the `shell-stack` as the root view.